gemrb/core/PathFinder.cpp

```diff
@@ -329,37 +360,16 @@ Path Map::FindPath(const Point& s, const Point& d, unsigned int size, unsigned i
 			if (childBlocked) continue;
 
 			// If there's an actor, check it can be bumped away
-			const Actor* childActor = GetActor(nmptChild, GA_NO_DEAD | GA_NO_UNSCHEDULED);
-			bool childIsUnbumpable = childActor && childActor != caller && (actorsAreBlocking || !childActor->ValidTarget(GA_ONLY_BUMPABLE));
+			const auto navmapCellTraversability = traversabilityCache.GetCellData(nmptChild.y * mapSize.w * 16 + nmptChild.x);
+			const bool childIsUnbumpable = navmapCellTraversability.occupyingActor != caller && navmapCellTraversability.state >= blockingTraversabilityValue;
 			if (childIsUnbumpable) continue;
 
 			SearchmapPoint smptCurrent2 { nmptCurrent };
 			NavmapPoint nmptParent = parents[smptCurrent2.y * mapSize.w + smptCurrent2.x];
 			SearchmapPoint smptParent { nmptParent };
 			unsigned short oldDist = distFromStart[smptChildIdx];
 
-			if (usePlainThetaStar) {
-				// Theta-star path if there is LOS
-				if (IsWalkableTo(nmptParent, nmptChild, actorsAreBlocking, caller)) {
-					unsigned short newDist = distFromStart[smptParent.y * mapSize.w + smptParent.x] + Distance(smptParent, smptChild);
-					if (newDist < oldDist) {
-						parents[smptChildIdx] = nmptParent;
-						distFromStart[smptChildIdx] = newDist;
-					}
-					// Fall back to A-star path
-				} else {
-					unsigned short newDist = distFromStart[smptCurrent2.y * mapSize.w + smptCurrent2.x] + Distance(smptCurrent2, smptChild);
-					if (newDist < oldDist) {
-						parents[smptChildIdx] = nmptCurrent;
-						distFromStart[smptChildIdx] = newDist;
-					}
-				}
-
-				if (distFromStart[smptChildIdx] < oldDist) {
-					PQNode newNode(nmptChild, getHeuristic(smptChild, smptChildIdx));
-					open.emplace(newNode);
-				}
-			} else {
+			{
```

Comment:
I appreciate this making the diff less noisy to review, but in the final version the extra scope should be removed. Can be an extra commit to maintain readability.

Reply:
Sure, it's left exactly for this purpose. It'll be removed.

Reply:
Done.